packages/cli-core/src/printer/printer.ts

```diff
@@ -4,27 +4,27 @@ import { EOL } from 'os';
 export type RecordValue = string | number | string[] | Date;
 
 /**
- * The class that pretty prints to the console.
+ * The class that pretty prints to the output stream.
  */
 export class Printer {
+  // Properties for ellipsis animation
+  private timer: ReturnType<typeof setTimeout>;
+  private timerSet: boolean;
+
   /**
-   * Print an object/record to console.
+   * Sets default configs
    */
-  static printRecord = <T extends Record<string | number, RecordValue>>(
-    object: T
-  ): void => {
-    let message = '';
-    const entries = Object.entries(object);
-    entries.forEach(([key, val]) => {
-      message = message.concat(` ${key}: ${val as string}${EOL}`);
-    });
-    console.log(message);
-  };
+  constructor(
+    private readonly minimumLogLevel: LogLevel,
+    private readonly stdout: NodeJS.WriteStream = process.stdout,
+    private readonly stderr: NodeJS.WriteStream = process.stderr,
+    private readonly refreshRate: number = 500
+  ) {}
 
   /**
-   * Prints an array of objects/records to console.
+   * Prints an array of objects/records to output stream.
    */
-  static printRecords = <T extends Record<string | number, RecordValue>>(
+  printRecords = <T extends Record<string | number, RecordValue>>(
     objects: T[]
```

Comment:
We could make this a spread input so you don't have to wrap a single entry in an array